File: 백준/Silver/10773. 제로/제로.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.Stack;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int k = Integer.parseInt(br.readLine());
        Stack<Integer> stack = new Stack<>();

        for (int i = 0; i < k; i++) {
            int input = Integer.parseInt(br.readLine());
            if (input != 0) {
                stack.push(input);
            } else {
                stack.pop();
            }
        }

        int sum = stack.stream().mapToInt(Integer::intValue).sum();
        StringBuilder sb = new StringBuilder();

        sb.append(sum);
        bw.write(sb.toString());
        bw.flush();
        bw.close();
    }
}

File: 백준/Silver/10773. 제로/제로.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.*

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val bw = BufferedWriter(OutputStreamWriter(System.out))
    val stack = Stack<Int>()

    val n = br.readLine().toInt()
    val sb = StringBuilder()

    repeat(n) {
        val input = br.readLine().toInt()

        if (input != 0) {
            stack.push(input)
        } else {
            stack.pop()
        }
    }

    val sum = stack.stream().mapToInt { it }.sum()
    sb.append(sum)

    bw.write(sb.toString())
    bw.flush()
    bw.close()
}